### PR TITLE
Hdrp/fix/shader stack normals

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
- Fixed null reference exception when using depth pre/post pass in shadergraph with alpha clip in the material: https://fogbugz.unity3d.com/f/cases/1253191/
- Fixed normal space in shadergraph templates

### Testing status
HDRP_Tests project is green locally.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/SurfaceOptionUIBlock.cs [hunk 1]
```diff
@@ -422,12 +422,14 @@ void DrawAlphaCutoffGUI()
                 {
                     if (transparentDepthPrepassEnable != null && transparentDepthPrepassEnable.floatValue == 1.0f)
                     {
-                        materialEditor.ShaderProperty(alphaCutoffPrepass, Styles.alphaCutoffPrepassText);
+                        if (alphaCutoffPrepass != null)
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/SurfaceOptionUIBlock.cs:425]: [optional] you can merge all the if

File: com.unity.render-pipelines.high-definition/Editor/Material/Lit/ShaderGraph/LitPass.template [hunk 2]
```diff
@@ -48,9 +48,6 @@ Pass
     $TransparentWritesMotionVec:        #define _WRITE_TRANSPARENT_MOTION_VECTOR 1
     $DepthOffset:                       #define _DEPTHOFFSET_ON 1
     $BlendMode.PreserveSpecular:        #define _BLENDMODE_PRESERVE_SPECULAR_LIGHTING 1
-    $NormalDropOffTS:                   #define _NORMAL_DROPOFF_TS 1
-    $NormalDropOffOS:                   #define _NORMAL_DROPOFF_OS 1
-    $NormalDropOffWS:                   #define _NORMAL_DROPOFF_WS 1
     $AttributesMesh.normalOS:           #define ATTRIBUTES_NEED_NORMAL
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Lit/ShaderGraph/LitPass.template:51]: Don't forget to also remove all the

        public static FieldDescriptor NormalDropOffTS =       new FieldDescriptor(string.Empty, "NormalDropOffTS", "_NORMAL_DROPOFF_TS 1");
        public static FieldDescriptor NormalDropOffOS =       new FieldDescriptor(string.Empty, "NormalDropOffOS", "_NORMAL_DROPOFF_OS 1");
        public static FieldDescriptor NormalDropOffWS =       new FieldDescriptor(string.Empty, "NormalDropOffWS", "_NORMAL_DROPOFF_WS 1");

File: com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/StackLitPass.template [hunk 3]
```diff
@@ -392,18 +392,9 @@ Pass
 
         // normal delivered to master node
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/StackLitPass.template:393]: you missed to remove $NormalDropOffTS:		                            #define _NORMAL_DROPOFF_TS 1 in stacklit

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/StackLitPass.template:393]: Do a pass on all _NORMAL_DROPOFF_TS / _NORMAL_DROPOFF_OS / _NORMAL_DROPOFF_WS in the code 
and NormalDropOffTS  / NormalDropOffOS / NormalDropOffWS

in C# and hlsl / template .shader

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/StackLit/ShaderGraph/StackLitPass.template:393]: oh universal use it, so only remove the thing that are on HDRP side

File: com.unity.render-pipelines.high-definition/Editor/Material/Lit/ShaderGraph/LitPass.template [hunk 4]
```diff
@@ -226,18 +223,9 @@ Pass
 
         // normal delivered to master node
         float3 normalSrc = float3(0.0f, 0.0f, 1.0f);
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Lit/ShaderGraph/LitPass.template:225]: normalSrc  is not used anymore (remark valid for all shader)

**Pull request merged**